### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by jasonscharf**
Adds comprehensive documentation around setup and usage.

### CODE REVIEW

**Comment by jasonscharf** [comment]
Notes:

- Best way to consume this PR, since it's primarily just a bunch of new content, is probably to navigate the branch and click through the Markdown links.

- I put the docs in /pages because I've got a follow up PR that allows the docs to be hosted live somewhere + syntax highlighting via next-mdx.

**Comment by tbakerx** [comment]
Hey @jasonscharf thanks for putting this together! Going through it now!

File: pages/docs/internals/how-it-works.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+
+## Experiment Lifecycle
+Experiments run for as long as they are considered active in the site config.
+
+When you feel that you've obtained statistical significance based on the observed metrics for the variants in a given experiment, you can mark the current experiment inactive, and promote the winning variant into permanent status by removing the `Variant` from around it.
+
+You should then mark the experiment as inactive like so:
+
+```json
```
Comment (tbakerx) [inline pages/docs/internals/how-it-works.md:282]: Should include `"inactive": true`?

File: pages/docs/internals/how-it-works.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+You can leave it in the site configuration as a historical
```
Comment (tbakerx) [inline pages/docs/internals/how-it-works.md:285]: Seems to trail off, should say something along the lines of "to maintain a history"?

File: pages/docs/setup/with-nextjs.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+- the `siteName` in case a non-default one is used
+- the `select` prop indicating with variant was selected in the current experiment
+- the `defer` flag, in case the component should perform its own variant selection client-side
+
+> **Note:** The SSR helper sets `defer` to `true` if the sessions or metrics backends are unavailable.
+> This forces client-side rendering in the case what SSR can't be fulfilled.
+
+Be sure to pass the props to `InstantBandit`:
+```tsx
+ <InstantBandit {...serverSideProps}>...</InstantBandit>
```
Comment (tbakerx) [inline pages/docs/setup/with-nextjs.md:77]: We should probably include the `Graceful Shutdown` nextjs-specific docs here in this setup?

**Review by tbakerx — CHANGES_REQUESTED** [review]
This is looking good! Thanks for putting this all together, should definitely help any new-comers to the repo get up and running. It does seem like there's a fair amount of duplication in the guides (setup/usage explained in several spots) however, more docs > less docs.

I've left a series of minor comments on just spelling/grammar stuff mostly. Have a look and then ping me and we can get er merged and released. Thanks again dude!

**Comment by jasonscharf** [comment]
@tbakerx You rock! Thanks so much for reviewing. I feel you on the duplication thing, it's a docs thing I've picked up over the years from tech writers and other examples (probably from reading too much docs lol). I might chop it down and bit, but I'll address the other items first. Thanks again!

**Comment by jasonscharf** [comment]
Hey @tbakerx , made some spot fixes based on your feedback. Let me know if I missed anything!

As for deduping some of the setup stuff, I'm going to hold off for now, unless you think it should be addressed in this PR. I'd like to remove the server setup steps entirely for the majority of users so only self-hosters need them. Beau gave me an idea there :)

**Pull request merged**